atomic_reactor/plugins/pre_inject_parent_image.py

```diff
@@ -0,0 +1,110 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import print_function, unicode_literals
+
+from atomic_reactor.build import ImageName
+from atomic_reactor.koji_util import create_koji_session
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.exit_remove_built_image import defer_removal
+from osbs.utils import graceful_chain_get
+
+
+class InjectParentImage(PreBuildPlugin):
+    """
+    Modifies parent image to be used based on given Koji build.
+
+    For each listed archive in Koji build, chose the archive that
+    defines a non-empty '.extra.docker.repositories' list.
```

Comment:
Maybe check build.extra.image.index.pull first? If that doesn't exist, the current approach is fine.